File: code/generate_strategies/IR(CoT)/java/java/function_itertools_cycle.java
package com.example;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;
import com.google.common.collect.Iterables;

public class CycleProcessor {
    
    public String test_cycle(String input_sequence, int num_elements) {
        
        if (input_sequence == null || input_sequence.isEmpty()) {
            return "";
        }
        
        List<Character> charList = new ArrayList<>();
        for (int i = 0; i < input_sequence.length(); i++) {
            charList.add(input_sequence.charAt(i));
        }
        
        Iterable<Character> cycleIterable = Iterables.cycle(charList);
        Iterator<Character> cycleIterator = cycleIterable.iterator();

        StringBuilder result = new StringBuilder();
        
        for (int i = 0; i < num_elements; i++) {
            result.append(cycleIterator.next());
        }
        return result.toString();
    }
}

File: code/test_suites/cpp/FunctionBuildTest/src/function_itertools_cycle.cpp
#include <string>
#include <vector>
#include <iterator>
#include <algorithm>

class CycleProcessor {
public:
    std::string test_cycle(const std::string& input_sequence, int num_elements) {
        if (input_sequence.empty()) {
            return "";
        }

        std::vector<char> charList(input_sequence.begin(), input_sequence.end());
        auto cycleIterator = charList.begin();

        std::string result;
        for (int i = 0; i < num_elements; ++i) {
            if (cycleIterator == charList.end()) {
                cycleIterator = charList.begin();
            }
            result += *cycleIterator++;
        }
        return result;
    }
};